core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -319,6 +323,18 @@ import scala.util.control.NonFatal
       sender() ! handleMetadataRequest(req)
   }
 
+  def expectSettings: Receive = LoggingReceive.withLabel("expectSettings") {
+    case s: ConsumerSettings[K, V] =>
+      applySettings(s)
+
+    case Stop =>
+      log.debug("Received Stop from {}, stopping", sender())
+      context.stop(self)
+
+    case _ =>
+      stash()
```

Comment:
Seems prudent to stash, but is it possible for the actor to receive an unhandled message in the lifetime it's using this receive?  Maybe an exception should be thrown instead since it's an unexpected state.

Reply:
The subscription messages immediately arrive here and need to be deferred until we have created the consumer.

Reply:
Ah, got it.